### ISSUE TITLE
Newline after jsdoc block suppresses validation

### ISSUE THREAD

**Issue by ndhoule** [eslint/eslint#2108]
ESLint checks this JSDoc comment as expected:

```js
/**
 * @param x
 * @param y
 * @param z
 * @return {undefined}
 */
var someFunction = function(x, y, z) {
  return x + y + z;
};
```

But adding a newline between the block comment and the function expression (also happens with function declarations) causes JSDoc validation to not happen:

```js
/**
 * @param x
 * @param y
 * @param z
 * @return {undefined}
 */

var someFunction = function(x, y, z) {
  return x + y + z;
};
```

Is this expected behavior?

**Comment by nzakas** [comment]
Yes. If the JSDoc isn't exactly above the function, we don't consider it. Otherwise, we could accidentally pull in JSDoc that has nothing to do with the function (such as with `@fileoverview` blocks).

**Comment by ndhoule** [comment]
Yeah, that's reasonable. I usually add a newline after a JSDoc comment, but an unscientific scan of GitHub repos shows I'm in the minority there, so I'm willing to concede that point.

The main problem I see with the current approach is exactly this case: Someone (inadvertently or otherwise) introduces a newline between a JSDoc comment block and the function, ESLint silently skips validation on that comment. As far as I can tell, there's no alternative means of preventing or detecting this situation without resorting to e.g. enforcing no empty lines.

One alternative approach I'm thinking about:

- Check the block comment to see if it contains a JSDoc-style `@tag` and only consider those for validation.
- When considering a comment for validation, require that the comment contain at least one of Constellation's recognized [rules][].

I don't like the silent/undetectable failure, but I'm not totally satisfied with this approach, either. Any ideas?

[rules]: https://github.com/Constellation/doctrine/blob/0835299b485ecdfa908d20628d6c8900144590ff/lib/doctrine.js#L596

**Comment by nzakas** [comment]
We have a suggestion for a `require-jsdoc` rule #1842, that might be the answer?

**Comment by gyandeeps** [comment]
@ndhoule any update on this?
Please reoen this issue if you want to continue the discussion this topic.
Thanks
